osquery/events/events.cpp

```diff
@@ -1049,8 +1052,8 @@ Status EventFactory::deregisterEventSubscriber(const std::string& sub) {
   }
 
   auto& subscriber = ef.event_subs_.at(sub);
-  subscriber->state(EventState::EVENT_NONE);
   subscriber->tearDown();
+  subscriber->state(EventState::EVENT_NONE);
```

Comment:
Why invert the order here?

Reply:
To be consistent with the other call sites that set state.